packages/react-hooks/src/hooks/timeout.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+
+export default function useTimeout(callback, delay: number) {
+  const callbackRef = React.useRef<() => void>();
+
+  React.useEffect(
```

Comment:
I don’t understand why this ref/ separate effect is needed

Reply:
Hmm I guess so you can change the callback and the original timeout will be respected? Imo that API aid not great, it’s better to just say they need to pass the same function, otherwise we cancel the timeout